src/client/visualizations/line-chart/base-chart/base-chart.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from "react";
+import { ReactNode } from "react";
+import { Stage } from "../../../../common/models/stage/stage";
+import { Unary } from "../../../../common/utils/functional/functional";
+import { Scale } from "../chart-line/chart-line";
+import { ContinuousTicks } from "../utils/pick-x-axis-ticks";
+import { ContinuousScale } from "../utils/scale";
+import { Background } from "./background";
+import getScale from "./scale";
+
+interface ChartLinesProps {
+  yScale: Scale;
+  lineStage: Stage;
+}
+
+class BaseChartProps {
+  children: Unary<ChartLinesProps, ReactNode>;
+  label: ReactNode;
+  xScale: ContinuousScale;
+  xTicks: ContinuousTicks;
+  chartStage: Stage;
+  formatter: Unary<number, string>;
+  yDomain: [number, number];
```

Comment:
what does yDomain mean?

Reply:
domain (min and max input values) for y-axis. it's common idiom in d3: yDomain, yScale ... etc.